### PR TITLE
adding links to first references of the vocabulary items

### PR DESCRIPTION
**Pull request by sanbrock**

### CODE REVIEW

**Comment by sanbrock** [comment]
If base class reference is the same as the first reference, do not duplicate the information and have the same link listed twice

**Comment by phyy-nx** [comment]
Discussion from Code Camp:

- Remove dependency on pynxtools, by adding the new function as new code either alongside or inside dev_tools/docs/nxdl.py
- Instead of a verbose link to the dependecy (e.g. `<NXxbase/entry>`, use `...` and a pop up tool tip that says "This field extends NXxbase/entry" with a link to NXxbase/entry.
- The above may require a new Sphinx subpage dependency.  That's ok.  Example: https://sphinx-hoverxref.readthedocs.io/en/latest/

**Comment by sanbrock** [comment]
Code Camp suggestions are implemented, the new format is  
definition: (required) [NX_CHAR](https://github.com/nexusformat/nxdl-types.html#nx-char) …[🔗](https://github.com/nexusformat/definitions/pull/NXxbase.html#nxxbase-entry-definition-field)  
with tootip of `override: NXxbase/entry/definition`  
Also the external dependency pynxtools is removed and the code is integrated.

**Review by PeterC-DLS — COMMENTED** [review]
Is it possible to refactor nexus.py to separate out the xml tree parsing and form another PR to add the `nexus_file` utility with its h5py dependency?

**Comment by PeterC-DLS** [comment]
Could you rebase this change off the newly refactored [method in nxdl_utils.py](https://github.com/FAIRmat-NFDI/pynxtools/blob/2d33a221d00f7c86c92d6d3321dc977566f47338/pynxtools/nexus/nxdl_utils.py#L676)?

**Comment by sanbrock** [comment]
yes, sure

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+requires-python = ""
+classifiers = [
+    "Operating System :: OS Independent"
+]
+dependencies = [
+    "lxml",
+    "pyyaml",
+    "sphinx>=5",
+    "sphinx-tabs",
+    "pytest",
```
Comment (PeterC-DLS) [inline pyproject.toml:23]: Add sphinx-toolbox?

File: .gitignore [hunk 2]
```diff
@@ -12,3 +12,23 @@ makelog.txt
 # Unknown
 /python/
 __github_creds__.txt
+
```
Comment (PeterC-DLS) [inline .gitignore:15]: Can these changes be removed?

File: MANIFEST.in [hunk 3]
```diff
@@ -0,0 +1,4 @@
+recursive-include applications/ *.nxdl.xml
```
Comment (PeterC-DLS) [inline MANIFEST.in:1]: Is this required

File: .gitignore (hunk 2 above)
Comment (sanbrock) [reply .gitignore:15]: yes it is removed in the subsequent PR #1782

File: pyproject.toml (hunk 1 above)
Comment (sanbrock) [reply pyproject.toml:23]: yes

**Review by PeterC-DLS — COMMENTED** [review]
When I build the python package, it has a version of `0.0.0`. Perhaps set it to something other than the default.

File: .gitignore (hunk 2 above)
Comment (PeterC-DLS) [reply .gitignore:15]: There's no such PR

File: dev_tools/utils/nxdl_utils.py [hunk 4]
```diff
@@ -0,0 +1,854 @@
+# pylint: disable=too-many-lines
```
Comment (PeterC-DLS) [inline dev_tools/utils/nxdl_utils.py:1]: This file has the same issues as mentioned in in-depth review performed in https://github.com/nexusformat/definitions/pull/1303/files#diff-42efc233d0fece860f874809d0c5b425905ff3341504af645d3f205b628cb5a4

**Comment by phyy-nx** [comment]
Superseded by #1303

**Pull request closed without merging**